app/src/androidTest/java/org/mozilla/focus/activity/CustomTabTest.java

```diff
@@ -0,0 +1,156 @@
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.SystemClock;
+import android.preference.PreferenceManager;
+import android.support.customtabs.CustomTabsIntent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiSelector;
+
+import junit.framework.Assert;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.R;
+
+import java.io.IOException;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static android.support.test.espresso.action.ViewActions.click;
+import static org.mozilla.focus.activity.TestHelper.waitingTime;
+import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;
+
+@RunWith(AndroidJUnit4.class)
+public class CustomTabTest {
+    private static final String MENU_ITEM_LABEL = "TestItem4223";
+    private static final String ACTION_BUTTON_DESCRIPTION = "TestButton1337";
+
+    private MockWebServer webServer;
+
+    private UiObject titleMsg = TestHelper.mDevice.findObject(new UiSelector()
+            .description("focus test page")
+            .enabled(true));
+
+    private UiObject actionButton = TestHelper.mDevice.findObject(new UiSelector()
+            .description(ACTION_BUTTON_DESCRIPTION)
+            .enabled(true));
+
+    private UiObject shareMenuitem = TestHelper.mDevice.findObject(new UiSelector()
+            .text("Share…")
+            .enabled(true));
+
+    private UiObject customMenuItem = TestHelper.mDevice.findObject(new UiSelector()
+            .text(MENU_ITEM_LABEL)
+            .enabled(true));
+
+    @Rule
+    public ActivityTestRule<MainActivity> activityTestRule  = new ActivityTestRule<>(
+            MainActivity.class, true, false);
+
+    @Test
+    public void testCustomTabUI() throws Exception {
+        try {
+            startWebServer();
+
+            // Launch activity with custom tabs intent
+            activityTestRule.launchActivity(createCustomTabIntent());
+
+            // Wait for website to load
+            Assert.assertTrue(titleMsg.waitForExists(waitingTime));
+
+            // Verify action button is visible
+            Assert.assertTrue(actionButton.waitForExists(waitingTime));
+
+            // Open the menu
+            TestHelper.menuButton.perform(click());
+
+            // Verify share action is visible
+            Assert.assertTrue(shareMenuitem.waitForExists(waitingTime));
+
+            // Verify custom menu item is visible
+            Assert.assertTrue(customMenuItem.waitForExists(waitingTime));
+
+            SystemClock.sleep(8000);
```

Comment:
Oops, this was only for testing locally - so that I had some time to interact with the UI myself. :)